tests/unit/lms/views/basic_lti_launch_test.py

```diff
@@ -156,11 +158,40 @@ def test_it_does_not_call_list_result_upsert_if_canvas(
 
         lis_result_sourcedid_service.upsert_from_request.assert_not_called()
 
+    @pytest.fixture
+    def frontend_app(self, patch):
+        return patch("lms.views.basic_lti_launch.frontend_app")
 
-class TestCanvasFileBasicLTILaunch(ConfiguredLaunch):
-    def make_request(self, context, pyramid_request):
-        BasicLTILaunchViews(context, pyramid_request).canvas_file_basic_lti_launch()
+    @pytest.fixture(autouse=True)
+    def via_url(self, patch):
+        return patch("lms.views.basic_lti_launch.via_url")
 
+    @pytest.fixture(autouse=True)
+    def lis_result_sourcedid_service(self, pyramid_config):
+        lis_result_sourcedid_service = mock.create_autospec(
+            LISResultSourcedIdService, instance=True, spec_set=True
+        )
+        pyramid_config.register_service(
+            lis_result_sourcedid_service, name="lis_result_sourcedid"
+        )
+        return lis_result_sourcedid_service
+
+    @pytest.fixture(autouse=True)
+    def lti_h_service(self, pyramid_config):
+        lti_h_service = mock.create_autospec(LTIHService, instance=True, spec_set=True)
+        pyramid_config.register_service(lti_h_service, name="lti_h")
+        return lti_h_service
+
+    @pytest.fixture
+    def LtiLaunches(self, patch):
+        return patch("lms.views.basic_lti_launch.LtiLaunches")
+
+    @pytest.fixture(autouse=True)
+    def ModuleItemConfiguration(self, patch):
+        return patch("lms.views.basic_lti_launch.ModuleItemConfiguration")
```

Comment:
All these fellas were hanging out at the bottom of the file, but they only apply to these tests.